src/compiler/sys.ts

```diff
@@ -1087,11 +1087,13 @@ namespace ts {
             }
 
             function readFile(fileName: string, _encoding?: string): string | undefined {
+                perfLogger.logStartReadFile(fileName);
                 if (!fileExists(fileName)) {
                     return undefined;

```

Comment:
This will miss the stop call .. instead of modifying this function how about calling this fun readFileWorker and and adding readFile to start and stop before returning result from the worker

Reply:
Good point. I made the change.